identity/manager.go

```diff
@@ -95,13 +100,99 @@ func (m *Manager) Create(ctx context.Context, i *Identity, opts ...ManagerOption
 	}
 
 	if err := m.r.PrivilegedIdentityPool().CreateIdentity(ctx, i); err != nil {
+		if errors.Is(err, sqlcon.ErrUniqueViolation) {
+			return m.generateCredentialHint(ctx, err, i)
+		}
 		return err
 	}
 
 	trace.SpanFromContext(ctx).AddEvent(events.NewIdentityCreated(ctx, i.ID))
 	return nil
 }
 
+func (m *Manager) generateCredentialHint(ctx context.Context, e error, i *Identity) (err error) {
+	var hints []string
+	var message string
+	for _, va := range i.VerifiableAddresses {
+		conflictingAddress, err := m.r.PrivilegedIdentityPool().FindVerifiableAddressByValue(ctx, va.Via, va.Value)
+		if errors.Is(err, sqlcon.ErrNoRows) {
+			continue
+		}
+		if err != nil {
+			m.r.Logger().WithError(err).Error("Failed to generate credential hint")
+			continue
+		}
+		message = fmt.Sprintf("An account for %s already exists.", va.Value)
+		hintIdentity, err := m.r.PrivilegedIdentityPool().GetIdentity(ctx, conflictingAddress.IdentityID, ExpandCredentials)
+		if err != nil {
+			m.r.Logger().WithError(err).Error("Failed to generate credential hint")
+			continue
+		}
+		for _, c := range hintIdentity.Credentials {
+			hints = append(hints, toPretty(&c)...)
+		}
+	}
+	if len(hints) == 0 {
+		hints = append(hints, "If you forgot your credentials, please recover your account from the sign-in page.")
+	} else {
+		hints = append(hints, "Alternatively, please recover your account from the sign-in page.")
+	}
+	return &ErrDuplicateCredentials{e, message, hints}
+}
+
+func toPretty(c *Credentials) (hints []string) {
+	switch c.Type {
+	case CredentialsTypeOIDC:
+		var cfg CredentialsOIDC
+		_ = json.Unmarshal(c.Config, &cfg)
+		for _, provider := range cfg.Providers {
+			hints = append(hints, fmt.Sprintf("Please sign in with %s instead.", provider.Provider))
+		}
+		return hints
+	case CredentialsTypeWebAuthn:
+		var cfg CredentialsWebAuthnConfig
+		_ = json.Unmarshal(c.Config, &cfg)
+		for _, webauthn := range cfg.Credentials {
+			if webauthn.IsPasswordless {
+				hints = append(hints, "Please sign passwordless using %s instead.", webauthn.DisplayName)
```

Comment:
Error message doesn't make sense. Can display name be empty? maybe we need a fallback?

```suggestion
				hints = append(hints, "Please sign in with passwordless on %s instead.", webauthn.DisplayName)
```